### PR TITLE
Enhance Orbax checkpointing documentation

### PR DESCRIPTION
**Pull request by cpgaffney1**
Add additional documentation about how to use pure Orbax alongside the existing Flax checkpointing documentation.

### CODE REVIEW

File: docs/guides/use_checkpointing.md [hunk 1]
```diff
@@ -1,34 +1,18 @@
...
-      extension: .md
-      format_name: markdown
-      format_version: '1.3'
-      jupytext_version: 1.13.8
-  kernelspec:
-    display_name: Python 3 (ipykernel)
-    language: python
-    name: python3
----
-
```
Comment (cgarciae) [inline docs/guides/use_checkpointing.md:15]: @cpgaffney1 we should keep the `jupytext` headers which as used for the `pre-commit` hooks. Please take a look at the [How to contribute](https://flax.readthedocs.io/en/latest/contributing.html) guide 🙂 .

Comment (cgarciae) [reply docs/guides/use_checkpointing.md:15]: TLDR; run:
```
pip install pre-commit
pre-commit install
pre-commit run --all-files
```
Else your PR will fail if the notebooks and markdown files are not in sync.

**Review by IvyZX — COMMENTED** [review]
Thank you Colin for improving the documentation!
(For context: Colin is the author of Orbax checkpointing.)

I was hoping to get rid of all the `flax.training.checkpoints` references in this doc and use only the Orbax ones, but that can be the content of another PR.

Please see Cristian's comment on fixing the doc build errors. Let me know if you need any help on it. After it gets fixed I will approve the PR and merge it.

File: docs/guides/use_checkpointing.md (hunk 1 above)
Comment (cpgaffney1) [reply docs/guides/use_checkpointing.md:15]: Thanks. I'm just going to make a new PR, merging is turning into more trouble than it's worth.

**Pull request closed without merging**